ui/app/components/pgp-list.js

```diff
@@ -1,23 +1,57 @@
-import { computed } from '@ember/object';
 import Component from '@ember/component';
 
 export default Component.extend({
   onDataUpdate: () => {},
-  listData: computed('listLength', function() {
-    let num = this.get('listLength');
+  listLength: 0,
+  listData: null,
+
+  init() {
+    this._super(...arguments);
+    let num = this.listLength;
     if (num) {
       num = parseInt(num, 10);
     }
-    return Array(num || 0)
+    let list = this.newList(num);
+    this.set('listData', list);
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    let list;
+    if (!this.listLength) {
+      this.set('listData', []);
+      return;
+    }
+    // no update needed
+    if (this.listData.length === this.listLength) {
+      return;
+    }
+    // shorten the current list
+    if (this.listLength < this.listData.length) {
+      list = this.listData.slice(0, this.listLength);
+    }
+    // add to the current list by creating a new list and copying over existing list
+    if (this.listLength > this.listData.length) {
```

Comment:
should this be an `else if`?

Reply:
I think that would be functionally the same, I just like keeping track of a single thing and this feels more like shallow`if` statements, but adding else isn't really another level. Let me update it!